### PR TITLE
[android] Use JSONObjectUtils in client code

### PR DESCRIPTION
**Pull request by wschurman**
# Why

These callsites can be greatly simplified using the new JSONObjectIUtils. Note that these utils only work in kotlin code since they are reified functions.

# How

Grep for `.optString(` and `.has(` and manually assess each case.

# Test Plan

Build and run NCL.

### CODE REVIEW

**Review by ide — APPROVED** [review]
I'm a bit uneasy about some of the type inference as opposed to explicitly specifying the generic types but I think this works.

File: android/expoview/src/main/java/host/exp/exponent/notifications/NotificationHelper.kt [hunk 1]
```diff
@@ -224,21 +221,9 @@ object NotificationHelper {
...
-        details.optString(NotificationConstants.NOTIFICATION_CHANNEL_PRIORITY)
-      } else {
-        null
-      }
-      val sound: Boolean? = if (!details.isNull(NotificationConstants.NOTIFICATION_CHANNEL_SOUND)) {
-        details.optBoolean(NotificationConstants.NOTIFICATION_CHANNEL_SOUND)
-      } else {
-        null
-      }
+      val description: String? = details.getNullable(NotificationConstants.NOTIFICATION_CHANNEL_DESCRIPTION)
```
[the remark this answers is not in the document]
Comment (wschurman) [reply android/expoview/src/main/java/host/exp/exponent/notifications/NotificationHelper.kt:224]: Yep, that's correct.

**Comment by wschurman** [comment]
> I'm a bit uneasy about some of the type inference as opposed to explicitly specifying the generic types but I think this works.

I was at first as well, and I thought about adding `getNullableString`, `getNullable[Type]`, etc methods. I ended up not doing so because of a few things that helped convince me that this pattern was safe:
1. If kotlin can't infer the reified type it errors, requiring the explicit type parameter.
2. If kotlin can infer the reified type it warns that the type parameter is extraneous.

The only downside is that it may be less clear in code review, especially since github isn't great at showing surrounding context, namely function signatures, which kotlin uses in a lot of cases to deduct the type.

**Comment by ide** [comment]
Given that we use Android Studio and can rely on inference, that sounds fine then.

**Pull request merged**